fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexScrubber.java

```diff
@@ -273,90 +267,24 @@ public ScrubbingPolicy build() {
      *
      */
     @API(API.Status.UNSTABLE)
-    public static class Builder {
-        @Nullable
-        protected FDBDatabaseRunner runner;
-        @Nullable
-        protected FDBRecordStore.Builder recordStoreBuilder;
+    public static class Builder extends OnlineIndexBaseBuilder<Builder> {
         @Nullable
         protected Index index;
         @Nullable
         protected Collection<RecordType> recordTypes;
-        @Nonnull
-        protected UnaryOperator<OnlineIndexer.Config> configLoader = old -> old;
 
         ScrubbingPolicy scrubbingPolicy = null;
         ScrubbingPolicy.Builder scrubbingPolicyBuilder = null;
 
 
-        protected int limit = 2000;
-        protected int maxWriteLimitBytes = OnlineIndexer.DEFAULT_WRITE_LIMIT_BYTES;
-        protected int maxRetries = OnlineIndexer.DEFAULT_MAX_RETRIES;
-        protected int recordsPerSecond = OnlineIndexer.DEFAULT_RECORDS_PER_SECOND;
-        private long progressLogIntervalMillis = OnlineIndexer.DEFAULT_PROGRESS_LOG_INTERVAL;
-        // Maybe the performance impact of this is low enough to be always enabled?
-        private boolean trackProgress = true;
-        private int increaseLimitAfter = OnlineIndexer.DO_NOT_RE_INCREASE_LIMIT;
         private long leaseLengthMillis = OnlineIndexer.DEFAULT_LEASE_LENGTH_MILLIS;
```

Comment:
The `leaseLengthMillis` seems to be belonged in the config instead of `IndexingCommon` (and be handled in the common builder). 
(It is used only once in the code, I think)

Reply:
Maybe? The problem you run into is trying to in structuring a change like this is that the lease length millis is ignored for the online indexer _only_ if it sets the "used synchonrized session parameter", whereas the scrubber always uses a syncrhonized session. The fact that this differs between the two implementations makes it a little hard to merge those configurations together. If we wanted to do this, I feel like we should also merge their behavior a bit more.

Reply:
I agree. If a non-synchronized indexing session can be useful in some cases, so should be a non-synchronized index scrubbing.